From 1d9bf6b69e4bdeecf23d00a1d083912ad479269e Mon Sep 17 00:00:00 2001
From: Li Shang <shangli1@huawei.com>
Date: Tue, 13 Nov 2018 22:43:48 +0800
Subject: [PATCH] huawei bugfix support ip route add vai

---
 ip/iproute.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/ip/iproute.c b/ip/iproute.c
index f252df4..2560b27 100644
--- a/ip/iproute.c
+++ b/ip/iproute.c
@@ -1467,9 +1467,6 @@ static int iproute_modify(int cmd, unsigned flags, int argc, char **argv)
 		argc--; argv++;
 	}
 
-	if (!dst_ok)
-		usage();
-
 	if (d) {
 		int idx = ll_name_to_index(d);
 
-- 
1.8.3.1


