From dff568963505f9f340a9da5151f1a022650e1498 Mon Sep 17 00:00:00 2001
From: wangli <wangli221@huawei.com>
Date: Tue, 3 Mar 2020 18:55:29 +0800
Subject: [PATCH] peer_notify_delay renamed to peer_notif_delay

---
 ip/iplink_bond.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/ip/iplink_bond.c b/ip/iplink_bond.c
index 585b6be..fb62c95 100644
--- a/ip/iplink_bond.c
+++ b/ip/iplink_bond.c
@@ -120,7 +120,7 @@ static void print_explain(FILE *f)
 		"Usage: ... bond [ mode BONDMODE ] [ active_slave SLAVE_DEV ]\n"
 		"                [ clear_active_slave ] [ miimon MIIMON ]\n"
 		"                [ updelay UPDELAY ] [ downdelay DOWNDELAY ]\n"
-		"                [ peer_notify_delay DELAY ]\n"
+		"                [ peer_notif_delay DELAY ]\n"
 		"                [ use_carrier USE_CARRIER ]\n"
 		"                [ arp_interval ARP_INTERVAL ]\n"
 		"                [ arp_validate ARP_VALIDATE ]\n"
@@ -166,7 +166,7 @@ static int bond_parse_opt(struct link_util *lu, int argc, char **argv,
 	__u8 xmit_hash_policy, num_peer_notif, all_slaves_active;
 	__u8 lacp_rate, ad_select, tlb_dynamic_lb;
 	__u16 ad_user_port_key, ad_actor_sys_prio;
-	__u32 miimon, updelay, downdelay, peer_notify_delay, arp_interval, arp_validate;
+	__u32 miimon, updelay, downdelay, peer_notif_delay, arp_interval, arp_validate;
 	__u32 arp_all_targets, resend_igmp, min_links, lp_interval;
 	__u32 packets_per_slave;
 	unsigned int ifindex;
@@ -201,11 +201,11 @@ static int bond_parse_opt(struct link_util *lu, int argc, char **argv,
 			if (get_u32(&downdelay, *argv, 0))
 				invarg("invalid downdelay", *argv);
 			addattr32(n, 1024, IFLA_BOND_DOWNDELAY, downdelay);
-		} else if (matches(*argv, "peer_notify_delay") == 0) {
+		} else if (matches(*argv, "peer_notif_delay") == 0) {
 			NEXT_ARG();
-			if (get_u32(&peer_notify_delay, *argv, 0))
-				invarg("invalid peer_notify_delay", *argv);
-			addattr32(n, 1024, IFLA_BOND_PEER_NOTIF_DELAY, peer_notify_delay);
+			if (get_u32(&peer_notif_delay, *argv, 0))
+				invarg("invalid peer_notif_delay", *argv);
+			addattr32(n, 1024, IFLA_BOND_PEER_NOTIF_DELAY, peer_notif_delay);
 		} else if (matches(*argv, "use_carrier") == 0) {
 			NEXT_ARG();
 			if (get_u8(&use_carrier, *argv, 0))
@@ -418,8 +418,8 @@ static void bond_print_opt(struct link_util *lu, FILE *f, struct rtattr *tb[])
 
 	if (tb[IFLA_BOND_PEER_NOTIF_DELAY])
 		print_uint(PRINT_ANY,
-			   "peer_notify_delay",
-			   "peer_notify_delay %u ",
+			   "peer_notif_delay",
+			   "peer_notif_delay %u ",
 			   rta_getattr_u32(tb[IFLA_BOND_PEER_NOTIF_DELAY]));
 
 	if (tb[IFLA_BOND_USE_CARRIER])
-- 
2.19.1

